src/lib.rs

```diff
@@ -647,6 +654,9 @@ pub mod tests {
                 &rands,
                 Some(rng),
             )?;
+
+            test_canonical_serialize_deserialize(true, &proof);
+
```

Comment:
Not related to the current changes:
Why is the test function called ``bad_degree_bound_test()``? 
And why doesn't it use segmentation (the sampled polynomials are only of supported_degree).